modules/social_features/social_group/modules/social_group_request/social_group_request.api.php

```diff
@@ -16,6 +16,11 @@
  * @param array $group_types
  *   List of group types used in open social.
  *
+ * @deprecated in social:11.1.0 and is removed from social:12.0.0. Use
```

Comment:
As you can see [here](https://git.drupalcode.org/project/social_course/-/merge_requests/22/diffs) it was used in the course module.

Reply:
I understand, and thank you for your clarification, but this is not covering open source users that might have implemented this hook. 

From our community perspective they are no longer able to hook into `hook_social_group_request_alter`, and this, unfortunately is a breaking change.

We had a similar case, [covered by this commit](https://github.com/goalgorilla/open_social/pull/2562/commits/bcd197aeb9a2bbc9395c88ae4916ec16aa6bcce6), when migrating from Open Social 11.

I will be happy to start a discussion onto better handle this :D

Reply:
What do you suggest - just remove this deprecated hook or using this again with a new hook or something else?

Reply:
Hi @OH17L, I would like to share some Drupal documentation on how to deprecate code, please below.

1. [Drupal deprecation policy](https://www.drupal.org/about/core/policies/core-change-policies/drupal-deprecation-policy)
2. [Hooks deprecation policy](https://www.drupal.org/about/core/policies/core-change-policies/drupal-deprecation-policy#how-hook)
3. [A change record](https://www.drupal.org/node/2881531) ( just one example of a change record), letting developers know what changed, and how they need to change their code in order to fit the new implementation.

That said, we could proceed with the following steps:

1. We need to make sure that `hook_social_group_request_alter` is still invoked, meaning that we need to add `\Drupal::moduleHandler()->alter('social_group_request', $group_types);` back to our code
2. From your implementation on [social_courses_advanced_request](https://git.drupalcode.org/project/social_course/-/merge_requests/22/diffs) I see that the structure of the data changed, and we need to convert the old hook structure to the new structure. ( Also we changed from an `alter` to a `invokeAll`)
3. I would suggest that we add the converting logic and the `alterDeprecated()` at `modules/social_features/social_group/src/JoinManager.php:: relations()`, if you see fit, since it's where the new hook is being invoked.

Please let me know your thoughts about this suggestions.